container.go

```diff
@@ -310,11 +377,73 @@ InfLoop:
 		case <-ctx.Done():
 			c.t.Fatalf("health check failure: %s", ctx.Err())
 		case <-time.After(1 * time.Second):
-			if err := c.healthcheck(); err != nil {
+			if err := c.healthcheck(ctx, c.cli, c); err != nil {
 				printf("(setup ) %-25s (%s) - container health failure: %s", c.Name, c.ID, err.Error())
 				continue InfLoop
 			}
 			break InfLoop
 		}
 	}
 }
+
+// wrapper around cli.ImagePull to fill ImagePullOptions with authentication information, if any.
+func (c *Container) imagePull(ctx context.Context) (io.ReadCloser, error) {
+	pullOptions := types.ImagePullOptions{}
+
+	// https://github.com/docker/distribution/blob/master/reference/reference.go#L7
+	//
+	// First part of the name *could* be a domain. If there is a corresponding entry in the
+	// .docker/config.json, it probably is.
+	//
+	// There is an undocumented hack to determine whether the first component is an actual domain, but it's
+	// shit: https://github.com/docker/distribution/blob/545102ea07aa9796f189d82f606b7c27d7aa3ed3/reference/normalize.go#L62
+	nameParts := strings.SplitN(c.ccfg.Image, "/", 2)
+
+	// get the credentials
+	if len(nameParts) >= 2 { // e.g.: quay.io/hans/myimage:latest
+		domain := nameParts[0]
+
+		token, err := getCredentialsFromConfig(domain)
+
+		// if err is non-nil, then we couldn't get credentials,
+		// because either it wasn't a domain or the user did not log in
+		if err == nil {
+			pullOptions.RegistryAuth = token
+		}
+	}
+
+	return c.cli.ImagePull(ctx, c.ccfg.Image, pullOptions)
+}
+
+// get credentials from ~/.docker/config.json
+func getCredentialsFromConfig(domain string) (string, error) {
+	cfg, err := clicfg.Load(clicfg.Dir())
+	if err != nil {
+		return "", err
+	}
+
+	dcfg, ok := cfg.AuthConfigs[domain]
+
+	if !ok {
+		return "", fmt.Errorf("Domain %s does not exist in config", domain)
```

Comment:
Violates linting: `error string should not be capitalized or end with punctuation`.